orchagent/intfsorch.cpp

```diff
@@ -1255,3 +1326,88 @@ void IntfsOrch::doTask(SelectableTimer &timer)
         }
     }
 }
+
+bool IntfsOrch::isRemoteSystemPortIntf(string alias)
+{
+    Port port;
+    if(gPortsOrch->getPort(alias, port))
+    {
+        return(port.m_system_port_info.type == SAI_SYSTEM_PORT_TYPE_REMOTE);
+    }
+    //Given alias is system port alias of the local port
+    return false;
+}
+
+bool IntfsOrch::voqSyncAddIntf(string &alias, sai_object_id_t &rif_id)
+{
+    //Sync only local interface. Confirm for the local interface and
+    //get the system port alias for key for syncing
+    Port port;
+    if(gPortsOrch->getPort(alias, port))
+    {
+        if(port.m_system_port_info.type == SAI_SYSTEM_PORT_TYPE_REMOTE)
+        {
+            return true;
+        }
+        alias = port.m_system_port_info.alias;
+    }
+    else
+    {
+        SWSS_LOG_ERROR("Port does not exist for %s!", alias.c_str());
+        return false;
+    }
+
+    //Get router interface to make sure it exists
+    sai_attribute_t attr;
+    sai_status_t status;
+
+    attr.id = SAI_ROUTER_INTERFACE_ATTR_PORT_ID;
+
+    status = sai_router_intfs_api->get_router_interface_attribute(rif_id, 1, &attr);
+    if (status != SAI_STATUS_SUCCESS)
+    {
+        SWSS_LOG_ERROR("Failed to get rif for %s:0x%lx, rv:%d", alias.c_str(), rif_id, status);
+        return false;
+    }
+
+    //Get the Real ID of the RIF ID
```

Comment:
Can you please explain why this is necessary?

Reply:
Since orchagent to syncd router interface create call is asynchronous, while trying to get the real id of router interface, we need to make sure that the router interface is actually created in SAI and real id is available in asic db. Attempt to retrieve any attribute of router interface object will suffice. Just need to make a GET call on the object to make sure the entry is available.

Reply:
Ok, "attempt to retrieve any attribute of router interface object will suffice" makes sense to me. Can we avoid posting the RIF ID of this interface to the CHASSIS_APP_DB because that number is meaningless in the other instances? It can lead to confusion. Please also add a comment here about this.

Reply:
The RIF ID is synced for future use when we have multiple vlan provisioning. O.k, for now, we'll avoid syncing RIF ID to CHASSIS_APP_DB. We'll also add comment about this.

Reply:
@eswaranb,

don't we require to have same rif_id (i.e vlan) when creating neighbors on the rif across all instances of it?.

Reply:
Yes, when we do support vlan interfaces, some ID has to be exchanged to the remote ASICs. However, that will not be this RIF_ID, because this is just a local ID that SAI gives us here and doesn't mean anything in the hardware. We will probably have to add something like a SYSTEM_VLAN (similar to SYSTEM_PORT and SYSTEM_LAG) to carry that hw ID to reference the vlan at that point.